puppet/lib/puppet/indirector/catalog/puppetdb.rb

```diff
@@ -119,11 +119,22 @@ def add_transaction_uuid(hash, transaction_uuid)
   # Include code_id in hash, returning the complete hash.
   #
   # @param hash [Hash] original data hash
-  # @param code_id [String] code_id
-  # @return [Hash] returns original hash augmented with transaction_uuid
+  # @return [Hash] returns original hash with a gaurunteed code_id key
+  # @api private
+  def add_code_id_if_missing(hash)
+    hash['code_id'] = hash['code_id']
```

Comment:
ensuring that the `original hash` has a `code_id` key, even if the value is nil.